SFML/SFML
# Add Wayland Support and refactor unix backends

----

## Description

This is a very much WIP PR to add Wayland support

This PR is related to the issue #1082 

## Tasks


## How to test this PR?

Default example should be enough to test window opening.

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```
